lms/djangoapps/courseware/migrations/0011_csm_id_bigint.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.23 on 2019-08-28 15:50
+from __future__ import unicode_literals
+
+import courseware.fields
+from django.db import migrations
+from django.db.migrations import AlterField
+
+class CsmBigInt(AlterField):
+    '''
+    Subclass AlterField migration class to split SQL between two different databases
+    We can't use the normal AlterField migration operation because Django generate and routes migrations at the model
+    level and the coursewarehistoryextended_studentmodulehistoryextended table is in a different database
+    '''
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        to_model = to_state.apps.get_model(app_label, self.model_name)
+        if schema_editor.connection.alias == 'student_module_history':
+            schema_editor.execute("ALTER TABLE `coursewarehistoryextended_studentmodulehistoryextended` MODIFY `student_module_id` bigint UNSIGNED NOT NULL;")
+        elif self.allow_migrate_model(schema_editor.connection.alias, to_model):
+            schema_editor.execute("ALTER TABLE `courseware_studentmodule` MODIFY `id` bigint UNSIGNED AUTO_INCREMENT NOT NULL;")
+
+    def database_backwards(self, app_label, schema_editor, from_state, to_state):
+        # Make backwards migration a no-op, app will still work if column is wider than expected
```

Comment:
I'm pretty sure that we wouldn't run the backwards, but I was curious if there was any possible reason that we would want to have the alter commands here in case something unexpected happened or just to avoid setting a precedent of having an empty backwards?

Reply:
@adzuci Without an empty backwards the backwards code in the parent class calls the forwards function with the inputs reversed, which in this case means it would just try to run the same ALTER again. This function could run the ALTER to change it back to ```int(11)```, but on a large table this is an expensive operation so I think it's best to avoid it. Generally people shouldn't be reversing this migration.